src/pages/en/reference/directives-reference.mdx

```diff
@@ -97,6 +97,8 @@ These directives control how [UI Framework components](/en/core-concepts/framewo
 
 By default, a UI Framework component is not hydrated in the client. If no `client:*` directive is provided, its HTML is rendered onto the page without JavaScript.
 
+A client directive must be placed on an imported component (i.e., not a [dynamic](/en/core-concepts/astro-components/#dynamic-tags) or [passed](/en/guides/markdown-content/#custom-components-with-imported-mdx) component).
```

Comment:
```suggestion
A client directive can only be used on a UI framework component that is directly imported into a `.astro` component.  Hydration directives are not supported when using [dynamic tags](/en/core-concepts/astro-components/#dynamic-tags) and [custom components passed via the `components` prop](/en/guides/markdown-content/#custom-components-with-imported-mdx).
```
Is it that they *must* be used on an imported component, or can *only* be used on this type of component?

I might go for something closer to this, and would give the exclusions a full, proper sentence. (I don't know if "directly imported" makes sense, but it was my first attempt at describing something that gets its own, explicit import statement.)

And, I noticed that both of those links go to places where our examples are *not* UI framework components. (e.g. it's an Astro component for Dynamic Tags). So, these types of components we're linking to aren't even guaranteed to be UI framework components in the first place, and I don't *think* that warrants any special reminder of that fact (i.e. if they're not UI framework, they totally can't have client directives)... but I did notice it so I'm mentioning it in case we think it's noteworth.

Reply:
I like this rewording and I think it helps remediate any confusion with the examples not being framework components!